test-result-summary-client/src/TabularView/TabularView.jsx

```diff
@@ -0,0 +1,703 @@
+    dateTransform(date, type) {
+        const dateSplit = date.split('/');
+        const jdkDate = dateSplit[2] + ((dateSplit[0].length < 2) ? "0" + dateSplit[0] : dateSplit[0]) + ((dateSplit[1].length < 2) ? "0" + dateSplit[1] : dateSplit[1]);
+
+        if (type === 'test') {
+            this.setState({testJdkDate: jdkDate});
+        } else {
+            this.setState({baselineJdkDate: jdkDate});
+        }
+    }
+    // Based on data returned from database, create relevant columns for platforms
+    generateColumns(platforms) {
+        const newArray = [];
+        let column = {
+        Header: 'Benchmark Name',
+        accessor: 'benchmarkName',
+        Cell: props => <span>{props.value.split(",")[0]} <br/> {props.value.split(",")[1]} <br/> {props.value.split(",")[2]}</span>
+        };
+
+        newArray.push(column);
+
+        for (let i = 0; i < platforms.length; i++) {
+            let platform = platforms[i];
+    	
+            column = {};
+            column.id = platform;
+            column.Header = platform.toUpperCase();
+            column.accessor = d => d.platforms[platform];
+            // Each cell needs to display the comparison by default, on hover displays further details
+            column.Cell = props => <Tooltip title={<div >
+                Raw Test Score: {this.handleProp(props.value, 'testScore')} <br/> 
+                Test CI: {this.handleProp(props.value, 'testCI')}  <br/>
+                Test JDK: {this.handleProp(props.value, 'testJdk')}	<br/> 
+                Test Sdk Resource: {this.handleProp(props.value, 'testSdkResource')}	<br/> 
+                Raw Baseline Score: {this.handleProp(props.value, 'baselineScore')} <br/> 
+                Basline CI: {this.handleProp(props.value, 'baselineCI')}  <br/>
+                Baseline JDK: {this.handleProp(props.value, 'baselineJdk')} <br/>
+                Baseline Sdk Resource: {this.handleProp(props.value, 'baselineSdkResource')}
+                </div> }>
+                <span onClick={() => this.handleLink(this.handleProp(props.value, 'buildUrl'))}> {this.handleProp(props.value, 'compare')} </span></Tooltip>;
+            column.getProps = (state, rowInfo) => (this.handleRegression(this.handleProp(rowInfo.row[platform], 'compare')));
+            newArray.push(column);
+        }
+        this.setState({columns:newArray, originalColumns: newArray});
+    }
+    // Set cell color based on comparison value
+    handleRegression (val) {
+        let color;
+        if (val <= 90) {
+            color = '#cc3232';}
+        else if (val <= 98) {
+            color =  '#F0F755';}
+        else if (val === 'N/A') {
+            color = 'grey';}
+        else if (val > 200) {
+            color = '#ffdbac';}
+        else color =  '#2dc937';
+
+        return {
+            style: {
+                fontSize: 25,
+                backgroundColor: color
+            }	
+        }
+    }
+
+    colorFilter(caller) {
```

Comment:
We should explain that we have 2 filters: benchmark filter and color filter. We always call the benchmark filter first, which uses the `originalData` to limit to the selected benchmarks. Then we call the `colorFilter`, which further limits the data based on relative comparison values.

Also, we should use a better name such as `firstFilter` than `caller` since it doesn't really tell its purpose. If `firstFilter` is true, then we need to call the benchmark filter first. If it's false, then it means that we've already done filtering based on benchmarks, and hence, we should apply the color filter now.